### ISSUE TITLE
Short form of event on Activities page

### ISSUE THREAD

**Issue by JortvD** [GEWIS/gewisweb#1804]
### What would you like?

The intent of showing a short form of events on the activities page is to display the large number of events that are important to promote, while decreasing the clutter and time to scroll to the bottom. See the example at the bottom how this can be displayed. There are a large number of events that can be good to show on the Activities page:

- Educational events -> deadlines for signing up to courses and resits, the BTA, the NSE and course evaluations
- Save the date for ie weekends, AMD and other days
- Other events like the Career Day, Wervingsdagen activities and Go Green Week

All of these events will only need to display: the date (some maybe a starting and end date), a title and a description. These events do not need to link to or have their own page, the intent is that there only feature is that they are listed on the Activities page. They will be controlled by the Board in the admin panel and so there will not be a creation form accessable to the general member base. The events also do not have to be visible in the Activity Archive.

![image](https://github.com/GEWIS/gewisweb/assets/10116429/f731239f-c87b-4c69-a37b-609b3f050a52)

### Other information

Optionally the Activities page can be renamed to Agenda or Upcoming to show that it also includes these events. I don't think this is necessary.

My implementation approach would be:

- Make an Event a distinct model. Maybe abstract with an Activity (so Event has subtypes ShortEvent and Activity), but I don't think it is necessary.
- Display Events on the activity/list page
- Create admin page as admin-event
- Possibly add support for tags like Career (or a future Education tag)

I would like to know what you think! And I am interested in implementing it myself.

**Comment by tomudding** [comment]
Something that may be interesting to do with this is to automate it based on the GEWIS calendar (Google). Perhaps we can add a specific tag or something to these events in that calendar and then automatically parse it every X hours to update the list and use that list to display them in the website. That way we prevent duplication of these "events" and ensure that they are always up-to-date.

**Comment by JortvD** [comment]
That would indeed be a good other way to implement to administrative side of this feature. I guess than the parsing would be done on the server side, which would than cache this somehow?

**Comment by tomudding** [comment]
> That would indeed be a good other way to implement to administrative side of this feature. I guess than the parsing would be done on the server side, which would than cache this somehow?

Yeah that is what I had in mind. Preferably it then goes into memcache (or redis) instead of the database. However, we will have to look into preventing cache slam.

I also do not know how this will work with longer 'events', as we have never come to a good solution for GH-732.

**Comment by tomudding** [comment]
We once again have this issue:

<img width="817" height="612" alt="Image" src="https://github.com/user-attachments/assets/905e8bb7-398c-498c-ae50-f27d2398235e" />

Which means that the next actual activity is never shown on the front page or in the quick overview.

I think that together with GH-2052, we can finally make this and GH-732 work. With proper categories, we do not have to determine a threshold in terms of days, but we can make it so that specific categories are always considered _continuous_ and thus "condensed" in the overview.
